core/controllers/base.py

```diff
@@ -688,7 +693,14 @@ def handle_exception(self, exception, unused_debug_mode):
             # For GET requests, there is no payload, so we check against
             # GET_HANDLER_ERROR_RETURN_TYPE.
             # Otherwise, we check whether self.payload exists.
-            if (self.payload is not None or
+
+            # This check is to avoid throwing of 401 when payload doesn't
+            # exists and self.payload is replaced by ThrowGetError object.
+            # TODO(#13155): Change this to self.normalized_payload
+            #  once schema is implemented for all handlers.
+            payload_exists = (self.payload is not None and not isinstance(
+                self.payload, utils.ThrowGetError))
+            if (payload_exists or
```

Comment:
```suggestion

            # This check is to avoid throwing of 401 when payload doesn't
            # exists and self.payload is replaced by ThrowGetError object.
            # TODO(#13155): Change this to self.normalized_payload
            # once schema is implemented for all handlers.
            payload_exists = (
                self.payload is not None and 
                not isinstance(self.payload, utils.ThrowGetError)
            )
            if (
                    payload_exists or
```

Reply:
done